subsys/bluetooth/host/gatt.c

```diff
@@ -3464,7 +3464,10 @@ static uint8_t disconnected_cb(const struct bt_gatt_attr *attr, uint16_t handle,
 bool bt_gatt_is_subscribed(struct bt_conn *conn,
 			   const struct bt_gatt_attr *attr, uint16_t ccc_type)
 {
-	const struct _bt_gatt_ccc *ccc;
+	uint16_t ccc_bits;
+	uint8_t properties;
+	uint8_t ccc_bits_encoded[sizeof(ccc_bits)];
```

Comment:
This could in theory also be
```suggestion
	uint16_t ccc_bits_encoded;
```

where you then can do `ccc_bits = sys_le16_to_cpu(ccc_bits_encoded)`

Either way works though, but this would make it more similar to how `properties` is handled

Reply:
I prefer the array for a practical reason and a simplicity (aesthetic) reason:
 - The array cannot be confused with the parsed value.
 - `sys_le16_to_cpu` is not a nice mathematical function. It takes in a number, and what number comes out depends on the CPU architecture. This is weird. `sys_get_le16` on the other hand always does the same thing.
 - `uint16_t ccc_bits_encoded` has the same problem as above. It will have different values on different platforms for the same meaning.

Reply:
Based on that, are you still OK with `properties` not being an array then? Like `uint8_t properties[1]`? 

> sys_le16_to_cpu is not a nice mathematical function. It takes in a number, and what number comes out depends on the CPU architecture. This is weird. sys_get_le16 on the other hand always does the same thing.

`sys_get_le16` works the same way as `sys_le16_to_cpu`; take a little-endian value and convert it to host endianess. I don't understand why you prefer one over the other. `sys_le16_to_cpu` can possibly be considered slightly better in terms of performance, as it is a no-op on little-endian archs, where `sys_get_le16` always does the operation.

Reply:
^_^

Not strong preferences from me on this topic, though I am generally in favor of consistency :) Fine to keep as is